dex/farm/src/base_functions.rs

```diff
@@ -236,15 +240,17 @@ where
         sc: &<Self as FarmContract>::FarmSc,
         caller: &ManagedAddress<<<Self as FarmContract>::FarmSc as ContractBase>::Api>,
         token_attributes: &<Self as FarmContract>::AttributesType,
-        farm_token_amount: BigUint<<<Self as FarmContract>::FarmSc as ContractBase>::Api>,
     ) -> BigUint<<<Self as FarmContract>::FarmSc as ContractBase>::Api> {
         if &token_attributes.original_owner != caller {
             sc.update_energy_and_progress(caller);
+        }
 
+        let user_total_farm_position_mapper = sc.user_total_farm_position(caller);
+        if user_total_farm_position_mapper.is_empty() {
             return BigUint::zero();
         }
 
-        sc.claim_boosted_yields_rewards(caller, farm_token_amount)
+        sc.claim_boosted_yields_rewards(caller, user_total_farm_position_mapper.get())
```

Comment:
Perform only one instead of two storage operations:

let user_total = sc.user_total_farm_position(caller).get();
if user_total == 0u32 {
   return BigUint::zero();
}

sc.claim_boosted_yields_rewards(caller, user_total);